### DISCUSSION TITLE
Reputation system for player interaction

### DISCUSSION THREAD

**Discussion by PyroTech03** [AscensionGameDev/Intersect-Engine#2113] [Ideas]
Goal: Introduce a system where players and guilds can set a reputation for other players and guilds.

Implementation: Allowing a setting of - 10 to +10 and zero being neutral. A player can set an individual or guild to any rating in this range. A guild officer+ can set this rep for the guild. These numbers can be color represented to relay the information to the player.

Use of this can allow players to signal good and bad relations to other players. This can be used for guilds to signal enemies, players to signal grefing/trolling players, and allow the playerbase to self govern.

Later this can be expanded to allow for things such as pets or owned defenses to respect the owners reputation list for deciding aggression as the event system gets expanded upon.

For any ORPG emphasizing social interaction, this is a crucial element. Not so much a traditional or PvE focused one.

**Comment by Arufonsu** [comment]
could be something similar to what MS uses for "Fame"
it's probably not that hard to implement, we already have target boxes for players there, we just need an extra player db value to add over though

![imagen](https://github.com/AscensionGameDev/Intersect-Engine/assets/17498701/dcdaadf6-9be3-4fff-ba75-79433e848736)

**Reply by PyroTech03** [reply]
That's different than what I was going for here. The best example is Eve Online.

Like if me and you meet in game. I can set a reputation value for you, from myself. This way in the future if I see you, I can see I set you as say a 5 on a scale of - 10 to +10. I liked our interaction and would like to work with you in the future.

On the flipside, say you think I overcharged you for an item or service. You set me as - 5 to know in the future you don't particularly care for me.

This should carry to guild leaders being able to set rep for players/guilds as well. You set a guild or player as a 10 to your guild, that entity would be considered an ally. And you inherit rep tables from your guild on top of what you set so you can see who your guild likes or hates.

This is strictly a social mechanic for player entities to other player entities. There's nothing stopping you from setting everyone to - 10 if you wanted, for example.

Fame to the way I understand it in MS would be like how well known you are to say an NPC or NPC faction to affect what quests they might give you.
